draft-ietf-webtrans-http3.md

```diff
@@ -197,6 +197,26 @@ The `webtransport` HTTP Upgrade Token uses the Capsule Protocol as defined in
 HTTP-DATAGRAM}} is not required by WebTransport and can safely be ignored by WebTransport
 endpoints.
 
+## Subprotocol Negotiation
+
+WebTransport over HTTP/3 offers a subprotocol negotiation mechanism, similar to
+TLS Application-Layer Protocol Negotiation Extension (ALPN) {{?RFC7301}}; the
+intent is to simplify porting pre-existing protocols that use QUIC and rely on
+this functionality.
+
+The user agent MAY include a `WebTransport-Subprotocols-Available` header field
+in the CONNECT request, enumerating the possible subprotocols; if received, the
+server MUST include a `WebTransport-Subprotocol` field with the subprotocol
+chosen in the 2xx response, or respond with an error code instead.
```

Comment:
This MUST makes this feature mandatory for servers (as opposed to optional for clients). I'd suggest keeping the optionality, similar to how ALPN works. For example:
```suggestion
in the CONNECT request, enumerating the possible subprotocols. If the server
receives such a header, it MAY include a `WebTransport-Subprotocol` field in
a successful (2xx) response. If it does, the server SHALL include a single
subprotocol from the client's list in that field. Servers MAY reject the request
if the client did not include a suitable subprotocol.
```

Reply:
Well, it's only mandatory if the client already expects the server to support it.

Reply:
Feels like we ought to say something about the case where the client asked for "foo, bar" and the server says "200" but doesn't include the header.